org.thecrossingchurch.CustomJobs/Jobs/HubspotIntegrationPatching.cs

```diff
@@ -0,0 +1,442 @@
+                        properties.Add( new HubspotPropertyUpdate() { property = "phone", value = mobile.NumberFormatted } );
+                    }
+                    else
+                    {
+                        properties.Add( new HubspotPropertyUpdate() { property = "phone", value = "" } );
+                    }
+
+                    // Handle Email
+                    string email = person.Email == "" ? "" : person.Email.ToStringSafe();
+                    if (email != "" )
+                    {
+                        email = email != null ? person.Email.ToLower() : "";  //person.Email.ToLower().ToStringSafe();
+                    }
+                    string emailtld = "";
+                    if (email.Contains("."))
+                    {
+                        string[] emailSep = email.Split( '.' );
+                        emailtld = "." + emailSep[emailSep.Length - 1];
+                    }
+                    else
+                    {
+                        email = "";
+                    }
+
+                    if ( person.CanReceiveEmail( true ) && hsPersonEmails.Contains( email ) == false && email.Contains( "@" ) && tlds.Contains( emailtld ) )
+                    {
+                        properties.Add( new HubspotPropertyUpdate() { property = "email", value = email } );
+                    }
+                    else
+                    {
+                        properties.Add( new HubspotPropertyUpdate() { property = "email", value = "" } );
+                    }
+
+                    // eNews Subscriber true or false
+                    string eNewsSub = ( person.Email != "" && person.Email != null && eNewsEmails.Contains( person.Email.ToLower() ) ) ? "true" : "false";
+                    properties.Add( new HubspotPropertyUpdate() { property = "enews_subscriber", value = eNewsSub } );
+                    Debug.WriteLine( "Patching: eNews: " + eNewsSub + " | " + person.Email );
+
+                    // Discpleship Step Path step completed date gathering
+                    Dictionary<int, string> stepsPath = new Dictionary<int, string>()
+                    {
+                        { 26, "baptism" },
+                        { 27, "life_groups" },
+                        { 29, "first_time_serving" }
+                    };
+                    foreach ( KeyValuePair<int, string> kvp in stepsPath )
+                    {
+                        var stepQuery = from s in _context.Steps
+                                       join pa in _context.PersonAliases on s.PersonAliasId equals pa.Id
+                                       where s.StepTypeId == kvp.Key && pa.Id == s.PersonAliasId && pa.PersonId == person.Id
+                                       select s.CompletedDateTime;
+                        var stepResult = stepQuery.FirstOrDefault().ToStringSafe();
+                        stepResult = stepResult != "" ? ConvertDate( ( DateTime ) stepResult.AsDateTime() ) : stepResult;
```

Comment:
This way of calling `ConvertDate` on `stepResult` is unnecessarily complex. Can you simplify it for the sake of readability?